### PR TITLE
Standardize on datetime over pendulum

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

This addresses a comment left by @sarayourfriend in https://github.com/WordPress/openverse-catalog/pull/632#discussion_r934777831

## Description

This PR standardizes all uses of a `datetime` or `datetime` related object/class on the standard library's `datetime` module rather than `pendulum`. Pendulum is a library that Airflow uses internally, but we don't specify it explicitly in our requirements. Additionally, the related `pendulum` analogs to `datetime` are actually subclasses of `datetime` objects, so it's perfectly safe to use the base classes everywhere as well.

## Testing Instructions

`just test` should be sufficient!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — COMMENTED** [review]
Nice :slightly_smiling_face: Looks like we were barely using pendulum anyway :sweat_smile:

**Comment by AetherUnbound** [comment]
Yea, I was just trying to conform with the Airflow standard on a few more recent PRs 😅

**Comment by AetherUnbound** [comment]
Ah shoot, it looks like we actually do need pendulum in some places:
```python
self = UtcDateTime(timezone=True)
value = datetime.datetime(2022, 8, 18, 21, 22, 22, 316222)
dialect = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7fc297b5b8e0>

    def process_bind_param(self, value, dialect):
        if value is not None:
            if not isinstance(value, datetime.datetime):
                raise TypeError('expected datetime.datetime, not ' + repr(value))
            elif value.tzinfo is None:
>               raise ValueError('naive datetime is disallowed')
```

I'll update for this specific case and add a comment.

**Comment by sarayourfriend** [comment]
If we do need to keep it around can we add it to the requirements?

**Comment by sarayourfriend** [comment]
If Airflow and the community use Pendulum I do think that's a pretty decent argument for using it. Especially if there are particular pitfall edge cases where Pendulum is required. It's a lot less mental overhead to just use one, either one, than get frustrated running into the edge cases suddenly (IMO). Especially if Pendulum is duck compatible with datetime.datetime for usages with libraries that don't know about pendulum.

**Review by obulat — APPROVED** [review]
LGTM!

**Comment by AetherUnbound** [comment]
> Especially if there are particular pitfall edge cases where Pendulum is required.

Pendulum is only required in a [test that explicitly runs an Airflow DAG directly](https://github.com/WordPress/openverse-catalog/pull/678/commits/676c48d5a6a6a418e34ce5f2375f02907253de46#diff-176b6052f4f4d99c946d6cd87df291c47af669878344996e7a49829c495bb446R68), which is not an step we'd perform as part of our normal operational code. Furthermore, while pendulum can serve as a drop-in replacement for `datetime.datetime`, there are [code changes necessary for using pendulum's `duration` over `datetime.timedelta`](https://pendulum.eustace.io/docs/#interval). We use `timedelta` a substantial amount, so this would incur a non-trivial code change:

```bash
# Number of uses of datetime.timedelta
$ rg -g "*.py" timedelta | wc -l
115
```

This PR constitutes a smaller change than standardizing on pendulum, and would likely lead to greater consistency with the wider Python contributor community (who are more likely to use the standard library's `datetime` over an optimized `pendulum`).

File: tests/dags/common/operators/test_postgres_result.py [hunk 1]
```diff
@@ -5,7 +5,6 @@
 import datetime
 import os
 
-import pendulum
```
Comment (krysal) [inline tests/dags/common/operators/test_postgres_result.py:8]: Nice to keep only one time library in this file, and standardize on `datetime` in general.

**Review by krysal — APPROVED** [review]

**Review by dhruvkb — APPROVED** [review]
Looks good to me. It's nice that we don't need to use an indirect dependency.
